src/wing/TiglWingHelperFunctions.cpp

```diff
@@ -0,0 +1,138 @@
+        cumulatedDepthDirection += dirDepth;
+    }
+    const CCPACSWingSegment& outerSegment = wing.GetSegment(wing.GetSegmentCount());
+    gp_XYZ dirDepth = outerSegment.GetChordPoint(1, 1).XYZ() - outerSegment.GetChordPoint(1, 0).XYZ();
+    dirDepth        = gp_XYZ(fabs(dirDepth.X()), fabs(dirDepth.Y()), fabs(dirDepth.Z()));
+    cumulatedDepthDirection += dirDepth;
+
+    switch (GetMajorDirection(wing)) {
+    case TIGL_Y_AXIS:
+        return cumulatedDepthDirection.X() >= cumulatedDepthDirection.Z() ? TiglAxis::TIGL_X_AXIS
+                                                                          : TiglAxis::TIGL_Z_AXIS;
+    case TIGL_Z_AXIS:
+        return cumulatedDepthDirection.X() >= cumulatedDepthDirection.Y() ? TiglAxis::TIGL_X_AXIS
+                                                                          : TiglAxis::TIGL_Y_AXIS;
+    case TIGL_X_AXIS:
+        return cumulatedDepthDirection.Z() >= cumulatedDepthDirection.Y() ? TiglAxis::TIGL_Z_AXIS
+                                                                          : TiglAxis::TIGL_Y_AXIS;
+    default:
+        return TiglAxis ::TIGL_X_AXIS;
+    }
+}
+
+TiglAxis GetMajorDirection(const CCPACSWing& wing)
+{
+    if (!HasShape(wing)) {
+        LOG(WARNING) << "CTiglWingHelper::GetMajorDirection: This wing has no shape -> impossible to determine the "
+                        "direction properly. The default direction will be returned";
+        return TIGL_Y_AXIS;
+    }
+
+    switch (wing.GetSymmetryAxis()) {
+    case TIGL_X_Y_PLANE:
+        return TiglAxis::TIGL_Z_AXIS;
+    case TIGL_X_Z_PLANE:
+        return TiglAxis::TIGL_Y_AXIS;
+    case TIGL_Y_Z_PLANE:
+        return TiglAxis ::TIGL_X_AXIS;
+    default:
+        // heuristic to find the best major axis
+        // first find the deep axis , then chose the major axis between the two left axis
+        gp_XYZ cumulatedSpanDirection(0, 0, 0);
+        gp_XYZ cumulatedDepthDirection(0, 0, 0);
+        for (int i = 1; i <= wing.GetSegmentCount(); ++i) {
+            const CCPACSWingSegment& segment = wing.GetSegment(i);
+            gp_XYZ dirSpan                   = segment.GetChordPoint(1, 0).XYZ() - segment.GetChordPoint(0, 0).XYZ();
+            gp_XYZ dirDepth                  = segment.GetChordPoint(0, 1).XYZ() - segment.GetChordPoint(0, 0).XYZ();
+            dirSpan  = gp_XYZ(fabs(dirSpan.X()), fabs(dirSpan.Y()), fabs(dirSpan.Z())); // why we use abs value?
+            dirDepth = gp_XYZ(fabs(dirDepth.X()), fabs(dirDepth.Y()), fabs(dirDepth.Z()));
+            cumulatedSpanDirection += dirSpan;
+            cumulatedDepthDirection += dirDepth;
+        }
+        const CCPACSWingSegment& outerSegment = wing.GetSegment(wing.GetSegmentCount());
+        gp_XYZ dirDepth = outerSegment.GetChordPoint(1, 1).XYZ() - outerSegment.GetChordPoint(1, 0).XYZ();
+        dirDepth        = gp_XYZ(fabs(dirDepth.X()), fabs(dirDepth.Y()), fabs(dirDepth.Z()));
+        cumulatedDepthDirection += dirDepth;
```

Comment:
I think it would be good to explain the intuition behind this heuristic somewhere, ideally in the docstring of the function. I understand that the cumulatedXXXDirections are the (scaled) average XXX directions of all segments?

Reply:
I actually don't know ;) (see comment below).

Reply:
It's not averaged but simply added. It simply checks in what direction the wing stretched most. I did not yet put that into the docstring but I mentioned, that it is a heuristic and should be replaced, when this is address in CPACS.